src-cpp/rdkafkacpp.h

```diff
@@ -1365,7 +1366,139 @@ class RD_EXPORT MessageTimestamp {
   int64_t timestamp;               /**< Milliseconds since epoch (UTC). */
 };
 
+/**
+ * @brief Headers object
+ *
+ * This object encapsulates the C implementation logic into a C++ object
+ * for use in RdKafka::Messages object.
+ */
+class RD_EXPORT Headers {
+ public:
+  virtual ~Headers() = 0;
+
+  /**
+   * @brief Header object
+   *
+   * This object represents a single Header with key value pair
+   * and an ErrorCode
+   *
+   * @remark dynamic allocation of this object is not supported.
+   *
+   */
+  class Header {
+   public:
+    Header(const std::string& key,
+           const char* value,
+           RdKafka::ErrorCode err = ERR_NO_ERROR):
+    key_(key), err_(err) {
+        value_container_.assign(value);
+    };
+    
+    std::string key() const {
+        return key_;
+    }
+
+    const char* value() const {
```

Comment:
`const char *value..`

Reply:
header values may be NULL, and are to be considered binary, so use const void * and a size_t length instead